scripts/test_metric_reporter/suite_reporter.py

```diff
@@ -0,0 +1,186 @@
+
+    def dict_with_fieldnames(self) -> dict[str, str | int | float | None]:
+        """Convert the test suite result to a dictionary with field names.
+
+        Returns:
+            dict[str, str | int | float | None]: Dictionary representation of the test suite result.
+        """
+        return {
+            "Date - Job Number": self.date_job,
+            "Date": self.date,
+            "Job Number": self.job,
+            "Status": self.status,
+            "Execution Time": self.execution_time,
+            "Success": self.success,
+            "Failure": self.failure,
+            "Skipped": self.skipped,
+            "Unknown": self.unknown,
+            "Total": self.total,
+            "Success Rate (%)": self.success_rate,
+            "Failure Rate (%)": self.failure_rate,
+            "Skip Rate (%)": self.skipped_rate,
+            "Unknown Rate (%)": self.unknown_rate,
+        }
+
+
+class SuiteReporterError(BaseError):
+    """Exception raised for errors in the SuiteReporter."""
+
+    pass
+
+
+class SuiteReporter:
+    """Handles the reporting of test suite results from CircleCI metadata."""
+
+    logger = logging.getLogger(__name__)
+
+    def __init__(self, test_metadata_directory: str) -> None:
+        """Initialize the reporter with the directory containing test metadata.
+
+        Args:
+            test_metadata_directory (str): Path to the directory with CircleCI test metadata.
+
+        Raises:
+            CircleCIJsonParserError: If there is an error parsing the test metadata.
+        """
+        self._test_metadata_directory = test_metadata_directory
+        self.results: list[SuiteReporterResult] = self._parse_circleci_metadata()
+
+    def _parse_circleci_metadata(self) -> list[SuiteReporterResult]:
+        results: list[SuiteReporterResult] = []
+        for job_test_metadata in CircleCIJsonParser.parse(self._test_metadata_directory):
+            started_at = datetime.strptime(job_test_metadata.job.started_at, DATETIME_FORMAT)
+            stopped_at = datetime.strptime(job_test_metadata.job.stopped_at, DATETIME_FORMAT)
+            execution_time = (stopped_at - started_at).total_seconds()
+            test_suite_result = SuiteReporterResult(
+                date=started_at.strftime(DATE_FORMAT),
+                job=job_test_metadata.job.job_number,
+                status=job_test_metadata.job.status,
+                execution_time=execution_time,
+            )
+
+            if not job_test_metadata.test_metadata:
+                continue
+
+            for test in job_test_metadata.test_metadata:
+                if test.result in SUCCESS_RESULTS:
+                    test_suite_result.success += 1
+                elif test.result == FAILURE_RESULT:
+                    test_suite_result.failure += 1
+                elif test.result == SKIPPED_RESULT:
+                    test_suite_result.skipped += 1
+                else:
+                    test_suite_result.unknown += 1
```

Comment:
Good spot for a `match`

Reply:
But... its so ugly... 

I get what you're saying though.

Reply:
😆 🤷🏻‍♀️ If you feel very strongly I'll change it